controllers/storagecluster/exporter.go

```diff
@@ -209,9 +265,515 @@ func CreateOrUpdateServiceMonitor(r *StorageClusterReconciler, instance *ocsv1.S
+		destSpec.Replicas = srcSpec.Replicas
+		destSpec.RevisionHistoryLimit = srcSpec.RevisionHistoryLimit
+		if destSpec.Selector == nil {
+			destSpec.Selector = &metav1.LabelSelector{}
+		}
+		if destSpec.Selector.MatchLabels == nil {
+			destSpec.Selector.MatchLabels = make(map[string]string)
+		}
+		for lKey, lValue := range srcSpec.Selector.MatchLabels {
+			destSpec.Selector.MatchLabels[lKey] = lValue
+		}
+		destSpec.Strategy = srcSpec.Strategy
+		if destSpec.Template.ObjectMeta.Labels == nil {
+			destSpec.Template.ObjectMeta.Labels = make(map[string]string)
+		}
+		for lKey, lValue := range srcSpec.Template.ObjectMeta.Labels {
+			destSpec.Template.ObjectMeta.Labels[lKey] = lValue
+		}
+		if len(destSpec.Template.Spec.Containers) == 0 {
+			destSpec.Template.Spec.Containers = make([]corev1.Container, 1)
+		}
+		destContainer0 := &destSpec.Template.Spec.Containers[0]
+		srcContainer0 := srcSpec.Template.Spec.Containers[0]
+		destContainer0.Args = srcContainer0.Args
+		destContainer0.Command = srcContainer0.Command
+		// if any environment variable is injected,
+		// we have to preserve that
+		for _, srcEnv := range srcContainer0.Env {
+			present := false
+			for _, dstEnv := range destContainer0.Env {
+				if dstEnv.Name == srcEnv.Name {
+					present = true
+					break
+				}
+			}
+			if !present {
+				destContainer0.Env = append(destContainer0.Env, srcEnv)
+			}
+		}
+		destContainer0.Image = srcContainer0.Image
+		destContainer0.Name = srcContainer0.Name
+		destContainer0.Ports = srcContainer0.Ports
+		destContainer0.SecurityContext = srcContainer0.SecurityContext
+		destContainer0.VolumeMounts = srcContainer0.VolumeMounts
+
+		destSpec.Template.Spec.ServiceAccountName = srcSpec.Template.Spec.ServiceAccountName
+		destSpec.Template.Spec.Volumes = srcSpec.Template.Spec.Volumes
+		return nil
+	}); err != nil {
+		r.Log.Error(err, "failed to create/update metrics exporter deployment", "NamespacedName", namespacedName)
+		return err
+	}
+	return nil
+}
+
+func createMetricsExporterServiceAccount(ctx context.Context, r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	expectedServiceAccount := corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      metricsExporterName,
+			Namespace: instance.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/component": metricsExporterName,
+				"app.kubernetes.io/name":      metricsExporterName,
+				"app.kubernetes.io/version":   "0.0.1",
+			},
+		},
+	}
+	currentServiceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      expectedServiceAccount.Name,
+			Namespace: expectedServiceAccount.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, currentServiceAccount, func() error {
+		if currentServiceAccount.Labels == nil {
+			currentServiceAccount.Labels = make(map[string]string)
+		}
+		for lKey, lValue := range expectedServiceAccount.Labels {
+			currentServiceAccount.Labels[lKey] = lValue
+		}
```

Comment:
We do not label ServiceAccounts today. We can remove these.

Reply:
Done.
Then we should be removing the labels from expected SA template as well, right?